themes/default/source/js/application.js

```diff
@@ -472,6 +472,30 @@ function validateSize(map) {
   });
 }
 
+/**
+* Translates the X-position of an element inside a container so that its contents
+* are contained
+*
+* @param {object} element to position
+* @param {object} container element
+* @param {number} container margin
+*/
+function setPositionInContainer(el, container, margin = 0) {
+  const elRect = el.getBoundingClientRect();
+  const containerRect = container.getBoundingClientRect();
+
+  const leftDiff = elRect.left - containerRect.left;
+  const rightDiff = containerRect.right - elRect.right;
+  const halfElWidth = elRect.width/2;
+  // x
+  if (rightDiff < 0) {
+    el.style.transform = `translateX(-${halfElWidth-rightDiff+margin}px)`;
+  } else if (leftDiff < 0) {
+    el.style.transform = `translateX(-${halfElWidth+leftDiff+margin}px)`;
+  }
```

Comment:
@hbalenda do we need to add a comment on `halfElWidth` that it is used because the element css selector already translates the position to be centered above an item? Should the sign on the formula be `-` when translating the element position to the left and to the right? For instance:
```suggestion
  if (rightDiff < 0) {
    el.style.transform = `translateX(-${halfElWidth-rightDiff-margin}px)`;
  } else if (leftDiff < 0) {
    el.style.transform = `translateX(+${halfElWidth+leftDiff+margin}px)`;
  }
```

Reply:
You're right, this was wrong. I changed the definitions of rightDiff and leftDiff to be positive if the element is out of bounds so it's easier to read and fixed the margin math.